compiler/AST/foralls.cpp

```diff
@@ -1599,6 +1600,10 @@ static ForallStmt* doReplaceWithForall(ForLoop* src)
   return dest;
 }
 
+bool shouldReplaceForLoopWithForall(ForLoop *forLoop) {
```

Comment:
(Still reading the code) is this a change to make the code more self documenting?

Reply:
Unless I skipped a detail, I find this function to be an overkill. `invokesParallelIterator` is a decent name in my view, too.

Reply:
I call this outside of `foralls.cpp` (in `functionResolution.cpp`) at a place where I need to know if something is going to later be converted to a forall, so I need a function in the header.

I suppose I could have just as well declared `invokesParallelIterator` in the header but I do feel that is less self documenting and if we change the conditions in `foralls.cpp` it's unlikely we'd remember to change them in `functionResolution.cpp`.

Reply:
OK, I am fine with that. Thanks!